assets/js/modules/analytics/datastore/setup-flow.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * `modules/analytics` data store: setup-flow.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { MODULES_ANALYTICS_4 } from '../../analytics-4/datastore/constants';
+
+const { createRegistrySelector } = Data;
+
+const baseSelectors = {
+	getSetupFlowMode: createRegistrySelector( ( select ) => (
+	) => {
+		if ( ! select( MODULES_ANALYTICS_4 ) ) {
+			return 'legacy';
+		}
+
+		const isAdminAPIWorking = select( MODULES_ANALYTICS_4 ).isAdminAPIWorking();
+
+		if ( isAdminAPIWorking === false ) {
+			return 'legacy';
+		}
+
+		if ( isAdminAPIWorking === undefined ) {
+			return undefined;
+		}
+
+		const accountID = select( STORE_NAME ).getAccountID();
+
+		if ( ! accountID ) {
+			return 'ua';
+		}
```

Comment:
Since yesterday I've realized that `undefined` means loading!  (Thanks to @felixarntz's JS Design Patterns doc for confirming that)

I'll revisit this section (and your comments) now